eng/restore/binplacePackages/binplacePackages.depproj

```diff
@@ -26,24 +26,40 @@
     </BinPlaceConfiguration>
   </ItemGroup>
 
-  <ItemGroup>
-    <!-- runtime dependency: System.Data.SqlClient uap10.0.16299 -->
-    <!-- runtime dependency: System.Diagnostics.PerformanceCounters netcoreapp2.0,net461 -->
-    <PackageReference Include="System.Memory" Condition="'$(TargetFramework)' == 'netcoreapp2.0' OR '$(TargetFramework)' == 'uap10.0.16299' OR '$(TargetsNetfx)' == 'true' OR ('$(TargetsNetStandard)' == 'true' AND '$(NETStandardVersion)' &gt;= 1.1)">
-      <Version>4.5.3</Version>
-    </PackageReference>
-
-    <PackageReference Include="System.Numerics.Vectors" Condition="'$(TargetsNetStandardLowerThan21)' == 'true' OR '$(TargetsNetfx)' == 'true'">
-      <Version>4.5.0</Version>
-    </PackageReference>
-
-    <PackageReference Include="System.Threading.Tasks.Extensions" Condition="'$(TargetFramework)' == 'netcoreapp2.0' OR '$(TargetsNetfx)' == 'true' OR '$(TargetsNetStandard)' == 'true'">
-      <Version>4.5.2</Version>
-    </PackageReference>
-
-    <!-- Only include the assets from the direct packages we reference in the output -->
-    <PackageToInclude Include="@(PackageReference)" />
-  </ItemGroup>
+  <Choose>
+    <!-- This is to make sure that nothing unintended gets restored for netcoreapp. -->
+    <When Condition="'$(TargetGroup)' != 'netcoreapp'">
+      <ItemGroup>
+        <!-- runtime dependency: System.Data.SqlClient uap10.0.16299 -->
+        <!-- runtime dependency: System.Diagnostics.PerformanceCounters netcoreapp2.0,net461 -->
+        <PackageReference Include="System.Memory" Condition="'$(TargetFramework)' == 'netcoreapp2.0' OR '$(TargetFramework)' == 'uap10.0.16299' OR '$(TargetsNetfx)' == 'true' OR ('$(TargetsNetStandard)' == 'true' AND '$(NETStandardVersion)' &gt;= 1.1)">
+         <Version>4.5.3</Version>
+        </PackageReference>
+
+        <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Condition="'$(TargetsNetStandard)' != 'true' OR '$(NETStandardVersion)' &gt;= 2.0">
+          <Version>1.0.0</Version>
+        </PackageReference>
+
+        <PackageReference Include="System.Numerics.Vectors" Condition="'$(TargetsNetStandardLowerThan21)' == 'true' OR '$(TargetsNetfx)' == 'true'">
+          <Version>4.5.0</Version>
+        </PackageReference>
+
+        <PackageReference Include="System.Threading.Tasks.Extensions" Condition="'$(TargetFramework)' == 'netcoreapp2.0' OR '$(TargetsNetfx)' == 'true' OR '$(TargetsNetStandard)' == 'true'">
+          <Version>4.5.2</Version>
+        </PackageReference>
+
+        <!-- Only include the assets from the direct packages we reference in the output -->
+        <PackageToInclude Include="@(PackageReference)" />
```

Comment:
I believe this should be common, move out of choose/when.

Reply:
I meant that PackageToInclude  should be common,.

Reply:
okay. we can still keep the AsyncInterface package in common ?

Reply:
You can, but a choose/when with only one when clause is the same thing as a condition on the itemgroup.  Maybe just reduce to that?  Trying to address @ViktorHofer's concern without making this thing too complex to reason about.